os/arch/arm/src/amebasmart/amebasmart_i2s.c

```diff
@@ -1809,3 +1812,86 @@ struct i2s_dev_s *amebasmart_i2s_initialize(uint16_t port)
 }
 #endif /* I2S_HAVE_RX || I2S_HAVE_TX */
 #endif /* CONFIG_AUDIO */
+
+#ifdef CONFIG_PM
+static void amebasmart_i2s_reinitialize(uint16_t port)
+{
+	struct amebasmart_i2s_config_s *hw_config_s = NULL;
+	struct amebasmart_i2s_s *priv;
+
+	/* Assign HW configuration */
+	if (port == I2S_NUM_2) {
+		hw_config_s = (struct amebasmart_i2s_config_s *)&amebasmart_i2s2_config;
+	} else if (port == I2S_NUM_3) {
+		hw_config_s = (struct amebasmart_i2s_config_s *)&amebasmart_i2s3_config;
+	} else {
+		i2serr("Please select I2S2 or I2S3 bus\n");
+		return NULL;
+	}
+
+	/* If the I2S structure is NULL, we should have faced some problem */
+	DEBUGASSERT(g_i2sdevice[port]);
+
+	priv = g_i2sdevice[port];
+
+	/* Get default configuration */
+	i2s_getdefaultconfig(priv);
+
+	/* I2s object reinitialization */
+	i2s_init(&priv->i2s_object, hw_config_s->i2s_sclk_pin, hw_config_s->i2s_ws_pin, hw_config_s->i2s_sd_tx_pin, hw_config_s->i2s_sd_rx_pin, hw_config_s->i2s_mclk_pin);
+
+	/* Initialize buffering */
+#if defined(I2S_HAVE_TX) && (0 < I2S_HAVE_TX)
+	i2s_set_dma_buffer(&priv->i2s_object, (char *)priv->i2s_tx_buf, NULL, I2S_DMA_PAGE_NUM, I2S_DMA_PAGE_SIZE); /* Allocate DMA Buffer for TX */
+#endif
+
+	/* configures IRQ */
+#if defined(I2S_HAVE_RX) && (0 < I2S_HAVE_RX)
+	priv->rx_isr_handler = (i2s_irq_handler)&i2s_transfer_rx_handleirq;
+#endif
+
+#if defined(I2S_HAVE_TX) && (0 < I2S_HAVE_TX)
+	priv->tx_isr_handler = (i2s_irq_handler)&i2s_transfer_tx_handleirq;
+#endif
+
+	ret = amebasmart_i2s_isr_initialize(priv);
+	if (ret != OK) {
```

Comment:
ret is not defined in this function, can you please update?

Reply:
sorry for mistake, it has been fixed.